src/AutoRest.CSharp/Common/Output/Models/Types/ClientOptionsTypeProvider.cs

```diff
@@ -3,12 +3,14 @@
 
 using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.Linq;
 
 namespace AutoRest.CSharp.Output.Models.Types
 {
     internal sealed class ClientOptionsTypeProvider : TypeProvider
     {
+        private static TextInfo TextInfo = new CultureInfo("en-us", false).TextInfo;
```

Comment:
I'm a little bit uncomfortable involving CultureInfo. Is there any specific reason? Could we implement this ourselves since it is a simple one. CultureInfo always has unpredictable pitfalls.

Reply:
I think it's OK since we've specified the language locale. And we know we use it to format codes, e.g. the service version which should be a combination of alphanumeric plus `-`.

Reply:
We only specify the locale. Locale cannot specify formats now. Is there any place we could confirm `-` must be explained as a delimiter?

Reply:
It's a typo, it should be underscore `_`.

Reply:
Thanks. I wasn't aware of `CultureInfo.InvariantCulture.TextInfo`.